test/jest/victory-line/victory-line.test.js

```diff
@@ -0,0 +1,304 @@
+        calculateD3Path(props, "line", 0)
+      );
+    });
+
+    it("renders the correct d3Shape path with custom interpolation function", () => {
+      const props = {
+        interpolation: curveCatmullRom,
+        scale: "linear",
+        padding: 50,
+        width: 400,
+        height: 300,
+        data: [
+          { x: 0, y: 0 },
+          { x: 1, y: 1 },
+          { x: 2, y: 2 }
+        ]
+      };
+
+      const { container } = render(<VictoryLine {...props} />);
+
+      expect(container.querySelector("path").getAttribute("d")).toEqual(
+        calculateD3Path(props, "line", 0)
+      );
+    });
+  });
+
+  describe("rendering with accessors", () => {
+    it("renders array-type data", () => {
+      const data = [
+        [1, 2],
+        [3, 4]
+      ];
+      const { container } = render(<VictoryLine data={data} />);
+      const lines = container.querySelectorAll("path");
+      expect(lines).toHaveLength(1);
+    });
+
+    it("renders data values with null accessor", () => {
+      const data = [1, 2, 3, 4];
+      const { container } = render(
+        <VictoryLine data={data} x={null} y={null} />
+      );
+      const lines = container.querySelectorAll("path");
+      expect(lines).toHaveLength(1);
+    });
+
+    it("renders deeply nested data", () => {
+      const data = [
+        { a: { b: [{ x: 1, y: 2 }] } },
+        { a: { b: [{ x: 3, y: 4 }] } }
+      ];
+      const { container } = render(
+        <VictoryLine data={data} x={"a.b[0].x"} y={"a.b.0.y"} />
+      );
+      const lines = container.querySelectorAll("path");
+      expect(lines).toHaveLength(1);
+    });
+
+    it("renders data ordered by x-value, by default", () => {
+      const data = [{ t: 0 /*x: 10, y: 1*/ }, { t: 1 /*x:  9, y: 1*/ }];
+
+      const { container } = render(
+        <VictoryLine
+          data={data}
+          x={({ t }) => 10 - t}
+          y={() => 1}
+          dataComponent={
+            <Curve data-json={(props) => JSON.stringify(props.data)} />
+          }
+        />
+      );
+
+      const line = container.querySelector("path");
+      const renderedData = JSON.parse(line.getAttribute("data-json"));
+
+      expect(renderedData[0].t).toEqual(1);
+      expect(renderedData[1].t).toEqual(0);
+    });
+
+    it("renders data ordered by value of sortKey, if given", () => {
+      const data = [{ t: 0 /*x: 10, y: 1*/ }, { t: 1 /*x:  9, y: 1*/ }];
+      const { container } = render(
+        <VictoryLine
+          data={data}
+          sortKey="t"
+          x={({ t }) => 10 - t}
+          y={() => 1}
+          dataComponent={
+            <Curve data-json={(props) => JSON.stringify(props.data)} />
+          }
+        />
+      );
+
+      const line = container.querySelector("path");
+      const renderedData = JSON.parse(line.getAttribute("data-json")).map(
+        ({ t }) => t
+      );
+
+      expect(renderedData).toEqual([0, 1]);
```

Comment:
This is kind of a hacky solution that allows us to test the rendered data, since we're no longer able to directly access props with Enzyme (for a good reason!). 

I might do a bit of research to see if there are better approaches to testing rendered data in data visualization, since an automated test suite can't easily test things like _is the line going up or down_ the same way we could test it visually.

Reply:
It's also worth asking whether we should still be testing things like data ordering at the integration test level, or whether we should just unit test the helper functions.